lib/plugin_helper.rb

```diff
@@ -0,0 +1,67 @@
+module RubyPluginHelper
+  # 'template' is the value of `target_mapping` from the inventory
+  # Returns an array of strings which are the lookups from the template
+  def config_lookup(template)
+    lookups = []
+    postwalk_vals(template) do |value|
+      # First collect the lookups to make
+      lookups << value if value.is_a?(String)
+      value
+    end
+    lookups.uniq!
+    lookups.map! { |lookup| lookup.split('.') }
```

Comment:
I dont follow why it is appears to be important to modify the lookups object in place. How come we cant do `lookups.uniq.map { |lookup| lookup.split('.') }`

Reply:
I think that's better